newsroom/wire/block_media/company_factory.py

```diff
@@ -0,0 +1,84 @@
+import time
+from flask import session, g
+from superdesk import get_resource_service
+
+
+class CompanyFactory:
```

Comment:
Instead of creating your own caching mechanism, you can use existing cache that we're already using (which is also used in this version of newsroom). [get_cached_resource_by_id](https://github.com/superdesk/newsroom/blob/master/newsroom/utils.py#L302).

Using the get_cached_resource_by_id means this entire module will no longer be needed

Reply:
I don't understand the reason here. Why create a new caching mechanism when we already have one that is used in Production for a long time now, and proven to work. I'm not saying this won't work, but we have something robust and very thoroughly tested.

What benefit does having a session level cache vs an app/redis cache?

Reply:
The get_cached_resource_by_id can work for sure as it has approved, but please review below reasons for this use case. much appreciate your support 

Moving from session-based to app-level caching in CompanyFactory would be a big code change and could cause problems. 

Session caching keeps sensitive permission data safe in user sessions and gives more control over caching for each admin user different to normal user and its cache 

The current setup works well and fits our needs, so it's probably better to stick with it than spend a lot of time changing things. as it is more than a general cache requirements, 

thanks and regards